virt_lightning/api.py

```diff
@@ -447,31 +449,67 @@ def redirect_request(self, req, fp, code, msg, hdrs, newurl):
     logger.debug("Date: %s", last_modified)
     size = r.headers["Content-Length"]
     logger.debug("Size: %s", size)
-    lenght = int(r.headers["Content-Length"])
+    length = int(size)
     chunk_size = MB * 1
     target_file = pathlib.PosixPath(
-        "{storage_dir}/upstream/{distro}.qcow2".format(
-            storage_dir=storage_dir, **kwargs
-        )
+        "{storage_dir}/upstream/{distro}.qcow2".format(**kwargs)
     )
     temp_file = target_file.with_suffix(".temp")
     if target_file.exists():
         logger.info("File already exists: %s", target_file)
         return
     with temp_file.open("wb") as fd:
-        while fd.tell() < lenght:
+        while fd.tell() < length:
             chunk = r.read(chunk_size)
             fd.write(chunk)
             if progress_callback:
-                progress_callback(fd.tell(), lenght)
+                progress_callback(fd.tell(), length)
     temp_file.rename(target_file)
     try:
         r = urllib.request.urlopen(
-            BASE_URL + "/images/{distro}/{distro}.yaml".format(**kwargs)
+            "{url}/{distro}.yaml".format(url=parent_url, **kwargs)
         )
     except urllib.error.HTTPError as e:
         if e.code == 404:
             pass
     with target_file.with_suffix(".yaml").open("wb") as fd:
         fd.write(r.read())
     logger.info("Image {distro} is ready!".format(**kwargs))
+
+
+def fetch(configuration=None, progress_callback=None, hv=None, **kwargs):
+    """
+    Retrieve a VM image from Internet.
+    """
+    if hv is None:
+        conn = libvirt.open(configuration.libvirt_uri)
+        hv = vl.LibvirtHypervisor(conn)
+        hv.init_storage_pool(configuration.storage_pool)
+
+    configuration = configuration if configuration is not None else Configuration()
```

Comment:
You can acutally write this:

`configuration = configuration or Configuration()`